.github/workflows/release.yml

```diff
@@ -6,48 +6,44 @@
 
 # Allow only one release to run at a time
 concurrency:
-  group: release-smithy-rs
+  group: release-smithy-rs-${{ inputs.dry_run }}
   cancel-in-progress: true
 
 env:
   rust_version: 1.62.1
 
 name: Release smithy-rs
-run-name: ${{ github.workflow }} - ${{ inputs.dry_run && 'Dry run' || 'Production run' }}
+run-name: ${{ github.workflow }} ${{ inputs.semantic_version }} (${{ inputs.commit_sha }}) - ${{ inputs.dry_run && 'Dry run' || 'Production run' }}
 on:
   workflow_dispatch:
     inputs:
+      commit_sha:
+        description: The SHA of the git commit that you want to release (e.g. b2318b0)
+        required: true
+        type: string
+      semantic_version:
+        description: The semver tag that you want to release (e.g. 0.52.1)
+        required: true
+        type: string
       dry_run:
-        description: Dry runs will only produce release artifacts, but will not cut a release tag in GitHub nor publish to crates.io
+        description: Dry runs will only produce release artifacts, but they will not cut a release tag in GitHub nor publish to crates.io
         required: true
         type: boolean
         default: true
 
 jobs:
-  main-branch-check:
```

Comment:
Is there any way we could check that the versions we will be publishing in `gradle.properties` aren't already published? Would be a nice sanity check that the `gradle.properties` was correctly updated.

Reply:
It'd be quite cumbersome.
Things can fail at any step, including publishing - leaving us with some unpublished runtime crates. I'm opting for building the workflow in such a way that all steps are idempotent, therefore allowing us to retry any transient failure by re-triggering the workflow from the right commit.